<launch>
    <arg name="host" default="localhost"/>
    <arg name="port" default="2000"/>
    <arg name="role_name" default="hero"/>
    <arg name="keyboard_control" default="False"/>
    <arg name="bag_dir" default="$(find awareness_detector)/output"/>
    <arg name="bag_prefix" default=""/>

    <arg name="pupil_ip" default="localhost"/>
    <arg name="pupil_port" default="50020"/>
    <arg name="screen_resolution_x" default="1920"/>
    <arg name="screen_resolution_y" default="1080"/>
    <arg name="visualize" default="True"/>

    <!-- global parameters -->
    <param name="/carla/host" value="$(arg host)"/>
    <param name="/carla/port" value="$(arg port)"/>
    <param name="/carla/role_name" value="$(arg role_name)"/>

    <!-- spawn ego vehicle -->
    <include file="$(find telecarla_manual_control)/launch/telecarla_manual_control.launch">
        <arg name="host" value="$(arg host)"/>
        <arg name="port" value="$(arg port)"/>
        <arg name="role_name" value="$(arg role_name)"/>
    </include>

    <include file="$(find gaze_detector)/launch/gaze_publisher.launch">
        <arg name="pupil_ip" value="$(arg pupil_ip)"/>
        <arg name="pupil_port" value="$(arg pupil_port)"/>
        <arg name="screen_resolution_x" value="$(arg screen_resolution_x)"/>
        <arg name="screen_resolution_y" value="$(arg screen_resolution_y)"/>
    </include>

    <include file="$(find awareness_detector)/launch/carla_awareness_model.launch">
        <arg name="camera_tf_name" default="/carla/$(arg role_name)/camera/rgb/front"/>
        <arg name="vehicle_info_topic" default="/carla/$(arg role_name)/vehicle_info"/>
        <arg name="visualize" value="$(arg visualize)"/>
    </include>

    <!-- rosbag -->
    <node pkg="rosbag" type="record" name="rosbag_record" args="-o $(arg bag_dir)/$(arg bag_prefix) /carla/objects /tf /awareness_detection/roi /gaze_publisher/gaze /gaze_publisher/fixation carla/$(arg role_name)/camera/rgb/front/camera_info /carla/marker /carla/lanes /carla/$(arg role_name)/lane_invasion /carla/$(arg role_name)/collision /carla/$(arg role_name)/vehicle_status /carla/$(arg role_name)/vehicle_info /carla/$(arg role_name)/camera/rgb/front/image_color"/>
</launch>
